packages/@react-spectrum/s2/src/Table.tsx

```diff
@@ -11,85 +11,1089 @@
+import {IconContext} from './Icon';
+// @ts-ignore
+import intlMessages from '../intl/*.json';
+import {LayoutNode} from '@react-stately/layout';
+import {LoadingState, Node} from '@react-types/shared';
+import {Menu, MenuItem, MenuTrigger} from './Menu';
+import {mergeStyles} from '../style/runtime';
+import {ProgressCircle} from './ProgressCircle';
+import {raw} from '../style/style-macro' with {type: 'macro'};
+import React, {createContext, ReactNode, useCallback, useContext, useMemo, useRef, useState} from 'react';
+import {Rect} from '@react-stately/virtualizer';
+import SortDownArrow from '../s2wf-icons/S2_Icon_SortDown_20_N.svg';
+import SortUpArrow from '../s2wf-icons/S2_Icon_SortUp_20_N.svg';
+import {useIsMobileDevice} from './utils';
+import {useLoadMore} from '@react-aria/utils';
+import {useLocalizedStringFormatter} from '@react-aria/i18n';
+
+interface S2TableProps {
+  /** Whether the Table should be displayed with a quiet style. */
+  isQuiet?: boolean,
+  /**
+   * Sets the amount of vertical padding within each cell.
+   * @default 'regular'
+   */
+  density?: 'compact' | 'spacious' | 'regular',
+  /**
+   * Sets the overflow behavior for the cell contents.
+   * @default 'truncate'
+   */
+  overflowMode?: 'wrap' | 'truncate',
+  /** How selection should be displayed. */
+  selectionStyle?: 'checkbox' | 'highlight',
+  // TODO: will we contine with onAction or rename to onRowAction like it is in RAC?
+  /** Handler that is called when a user performs an action on a row. */
+  onAction?: (key: Key) => void,
+  /**
+   * Handler that is called when a user starts a column resize.
+   */
+  onResizeStart?: (widths: Map<Key, ColumnSize>) => void,
+  /**
+   * Handler that is called when a user performs a column resize.
+   * Can be used with the width property on columns to put the column widths into
+   * a controlled state.
+   */
+  onResize?: (widths: Map<Key, ColumnSize>) => void,
+  /**
+   * Handler that is called after a user performs a column resize.
+   * Can be used to store the widths of columns for another future session.
+   */
+  onResizeEnd?: (widths: Map<Key, ColumnSize>) => void,
+  /** The current loading state of the table. */
+  loadingState?: LoadingState,
+  /** Handler that is called when more items should be loaded, e.g. while scrolling near the bottom. */
+  onLoadMore?: () => any
+}
+
+// TODO: Note that loadMore and loadingState are now on the Table instead of on the TableBody
+export interface TableProps extends Omit<RACTableProps, 'style' | 'disabledBehavior' | 'className' | 'onRowAction' | 'selectionBehavior' | 'onScroll' | 'onCellAction' | 'dragAndDropHooks'>, StyleProps, S2TableProps {
+}
+
+let InternalTableContext = createContext<TableProps & {scale?: Scale, layout?: S2TableLayout<unknown>, setIsInResizeMode?:(val: boolean) => void, isInResizeMode?: boolean}>({});
+
+const tableWrapper = style({
+  width: 'full'
+});
+
+const table = style<TableRenderProps & S2TableProps>({
+  height: 'full',
```

Comment:
does this still work if the parent is a `flex child` instead of explicitly setting a height?

Reply:
So if the parent has display flex and there are other flex children alongside the table that the height of the container gets distributed amongst?

Reply:
yeah, or a flex laid out horizontally and relying on align-items stretch

Reply:
I did some testing around this and it seems fine in the case where the flex direction is row but flex direction column was problematic since the table wouldn't respect the container's height (probably what you were alluding to here). Think I fixed it now, lemme know if the new stories reflect what you were getting at